[CLANG] Use of NaN is undefined behavior in RecoTracker/PixelSeeding

CLANG complains about usage of `std::isnan` in RecoTracker/PixelSeeding due to the code being compiler with `-ffast-math`:

```
In file included from src/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernels.cc:3:
  src/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernelsImpl.h:485:18: warning: use of NaN is undefined behavior due to the currently enabled floating-point options [-Wnan-infinity-disabled]
   485 |         isNaN |= std::isnan(tracks_view[it].state()(i));
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernels.cc:152:3: note: in instantiation of function template specialization 'caHitNtupletGeneratorKernels::kernel_classifyTracks<pixelTopology::Phase1>' requested here
  152 |   kernel_classifyTracks<TrackerTraits>(tracks_view, this->params_.qualityCuts_);
      |   ^
```

```
In file included from src/RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernels.dev.cc:20:
  src/RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernelsImpl.h:525:20: warning: use of NaN is undefined behavior due to the currently enabled floating-point options [-Wnan-infinity-disabled]
   525 |           isNaN |= std::isnan(tracks_view[it].state()(i));
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/invoke.h:61:14: note: in instantiation of function template specialization 'alpaka_serial_sync::caHitNtupletGeneratorKernels::Kernel_classifyTracks<pixelTopology::Phase1>::operator()<alpaka::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int>, void>' requested here
   61 |     { return std::forward<_Fn>(__f)(std::forward<_Args>(__args)...); }
      |              ^
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/alpaka/1.1.0-0a6641b4bfdf883c8da5b9e8620be504/include/alpaka/queue/QueueGenericThreadsBlocking.hpp:135:17: note: in instantiation of member function 'alpaka::TaskKernelCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int, alpaka_serial_sync::caHitNtupletGeneratorKernels::Kernel_classifyTracks<pixelTopology::Phase1>, reco::TrackSoA<pixelTopology::Phase1>::Layout<>::ViewTemplateFreeParams<128, false, true, true> &, const pixelTrack::QualityCutsT<pixelTopology::Phase1> &>::operator()' requested here
  135 |                 task();
      |                 ^
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/alpaka/1.1.0-0a6641b4bfdf883c8da5b9e8620be504/include/alpaka/queue/Traits.hpp:49:54: note: in instantiation of member function 'alpaka::trait::Enqueue<alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>, alpaka::TaskKernelCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int, alpaka_serial_sync::caHitNtupletGeneratorKernels::Kernel_classifyTracks<pixelTopology::Phase1>, reco::TrackSoA<pixelTopology::Phase1>::Layout<>::ViewTemplateFreeParams<128, false, true, true> &, const pixelTrack::QualityCutsT<pixelTopology::Phase1> &>>::enqueue' requested here
   49 |         trait::Enqueue<TQueue, std::decay_t<TTask>>::enqueue(queue, std::forward<TTask>(task));
      |                                                      ^
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/alpaka/1.1.0-0a6641b4bfdf883c8da5b9e8620be504/include/alpaka/kernel/Traits.hpp:312:9: note: in instantiation of function template specialization 'alpaka::enqueue<alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>, alpaka::TaskKernelCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int, alpaka_serial_sync::caHitNtupletGeneratorKernels::Kernel_classifyTracks<pixelTopology::Phase1>, reco::TrackSoA<pixelTopology::Phase1>::Layout<>::ViewTemplateFreeParams<128, false, true, true> &, const pixelTrack::QualityCutsT<pixelTopology::Phase1> &>>' requested here
  312 |         enqueue(queue, createTaskKernel<TAcc>(workDiv, kernelFnObj, std::forward<TArgs>(args)...));
      |         ^
src/RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernels.dev.cc:378:13: note: in instantiation of function template specialization 'alpaka::exec<alpaka::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int>, alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>, alpaka::WorkDivMembers<std::integral_constant<unsigned long, 1>, unsigned int>, alpaka_serial_sync::caHitNtupletGeneratorKernels::Kernel_classifyTracks<pixelTopology::Phase1>, reco::TrackSoA<pixelTopology::Phase1>::Layout<>::ViewTemplateFreeParams<128, false, true, true> &, const pixelTrack::QualityCutsT<pixelTopology::Phase1> &>' requested here
  378 |     alpaka::exec<Acc1D>(
      |             ^
```

--- a/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernelsImpl.h
+++ b/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernelsImpl.h
@@ -17,6 +17,8 @@
 #include "HeterogeneousCore/CUDAUtilities/interface/cuda_assert.h"
 #include "RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforGPU.h"
 
+#include "FWCore/Utilities/interface/isFinite.h"
+
 #include "CAHitNtupletGeneratorKernels.h"
 #include "CAStructures.h"
 #include "GPUCACell.h"
@@ -482,7 +484,7 @@ namespace caHitNtupletGeneratorKernels {
       // if the fit has any invalid parameters, mark it as bad
       bool isNaN = false;
       for (int i = 0; i < 5; ++i) {
-        isNaN |= std::isnan(tracks_view[it].state()(i));
+        isNaN |= edm::isNotFinite(tracks_view[it].state()(i));
       }
       if (isNaN) {
 #ifdef NTUPLE_DEBUG

--- a/RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernelsImpl.h
+++ b/RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtupletGeneratorKernelsImpl.h
@@ -21,6 +21,7 @@
 #include "HeterogeneousCore/AlpakaInterface/interface/AtomicPairCounter.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/config.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "FWCore/Utilities/interface/isFinite.h"
 
 // local includes
 #include "CACell.h"
@@ -522,7 +523,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::caHitNtupletGeneratorKernels {
         // if the fit has any invalid parameters, mark it as bad
         bool isNaN = false;
         for (int i = 0; i < 5; ++i) {
-          isNaN |= std::isnan(tracks_view[it].state()(i));
+          isNaN |= edm::isNotFinite(tracks_view[it].state()(i));
         }
         if (isNaN) {
 #ifdef NTUPLE_DEBUG

